src/TableView.jl

```diff
@@ -1,119 +1,155 @@
+    on(rowparams) do x
+        requestedrows[] = table2json(rows, names, types, requested = [x["startRow"], x["endRow"]])
     end
 
-    subt = NDSparse(showna(ks), showna(vs))
-
-    headers = colnames(subt)
-    cols = [merge(Dict(:data=>n), get(colopts, n, Dict())) for n in headers]
+    onjs(requestedrows, @js function (val)
+        ($rowparams[]).successCallback(JSON.parse(val), $(tablelength))
+    end)
 
     options = Dict(
-        :data => JuliaDB.rows(subt),
-        :colHeaders => headers,
-        :fixedColumnsLeft => ndims(t),
-        :modifyColWidth => @js(w -> w > 300 ? 300 : w),
-        :modifyRowHeight => @js(h -> h > 60 ? 50 : h),
-        :manualColumnResize => true,
-        :manualRowResize => true,
-        :columns => cols,
-        :width => 800,
-        :height => 400,
+        :columnDefs => coldefs,
+        :enableSorting => true,
+        :enableFilter => true,
+        :maxConcurrentDatasourceRequests => 1,
+        :cacheBlockSize => 1000,
+        :maxBlocksInCache => 100,
+        :enableColResize => true,
+        :multiSortKey => "ctrl",
+        :rowModelType => "infinite",
+        :datasource => Dict(
+            "getRows" =>
+                @js function (rowParams)
+                    $rowparams[] = rowParams
+                end
+            ,
+            "rowCount" => tablelength
+        )
     )
 
-    merge!(options, Dict(kwargs))
-
-    handler = @js function (Handsontable)
-        @var sizefix = document.createElement("style");
-        sizefix.textContent = """
-            .htCore td {
-                white-space:nowrap
-            }
-        """
-        this.dom.appendChild(sizefix)
-        this.hot = @new Handsontable(this.dom, $options);
+    handler = @js function (agGrid)
+        @var gridOptions = $options
+        this.table = @new agGrid.Grid(this.dom.querySelector("#grid"), gridOptions)
+        gridOptions.columnApi.autoSizeColumns($names)
     end
     onimport(w, handler)
-    w.dom = dom"div"()
-    w
 end
 
-showtable(t; kwargs...) = showtable(table(t); kwargs...)
+# directly write JSON instead of allocating temporary dicts etc
+function table2json(rows, names, types; requested = nothing)
+    io = IOBuffer()
+    print(io, '[')
+    for (i, row) in enumerate(rows)
+        if requested == nothing || first(requested) <= i <= last(requested)
+            print(io, '{')
+            i = 1
+            for col in Tables.eachcolumn(row)
+                JSON.print(io, names[i])
+                i += 1
+                print(io, ':')
+                if col isa Number
+                    JSON.print(io, col)
+                else
+                    JSON.print(io, sprint(print, col))
+                end
+                print(io, ',')
+            end
+            skip(io, -1)
+            print(io, '}')
+            print(io, ',')
+        end
+    end
+    skip(io, -1)
+    print(io, ']')
 
-end # module
+    String(take!(io))
+end
```

Comment:
I love this!

Reply:
The unfortunate thing is that I can't figure out how to transfer the data with WebIO without an additional `JSON.parse` on the javascript side. Any ideas on how to circumvent that?